### PR TITLE
Install game files from the base image

### PR DESCRIPTION
**Pull request by thasmo**

### CODE REVIEW

**Comment by thasmo** [comment]
Probably fails because the DST game files don't have the proper permissions set or something after they get copied from the base image.

**Comment by thasmo** [comment]
Setting the owner after copying the files seems to fix the issue although running the `chown -R $STEAM_USER:$STEAM_USER $DST_HOME` takes quite a while.

**Comment by thasmo** [comment]
Maybe it's possible to use `USER $STEAM_USER` instead of running the `chmod` command but that may have other effects.

**Comment by BraisGabin** [comment]
I definitely like the idea of splitting this images in two. But why do you use the multi-stage builds? What do we get from them?

File: build/Dockerfile [hunk 1]
```diff
@@ -33,22 +35,9 @@ ENV DESCRIPTION="Powered by DST Academy." \
...
-		+app_update 343050 \
-			$([ -n "$DST_BRANCH" ] && printf %s "-beta $DST_BRANCH") \
-			$([ -n "$DST_BRANCH_PASSWORD" ] && printf %s "-betapassword $DST_BRANCH_PASSWORD") \
-			validate \
-		+quit \
-	&& rm -rf $STEAM_HOME/Steam/logs $STEAM_HOME/Steam/appcache/httpcache \
-	&& find $STEAM_HOME/package -type f ! -name "steam_cmd_linux.installed" ! -name "steam_cmd_linux.manifest" -delete
+# Install Don't Starve Together.
+COPY --from=BASE /opt/dst $DST_HOME
+RUN chown -R $STEAM_USER:$STEAM_USER $DST_HOME
```
Comment (BraisGabin) [inline build/Dockerfile]: You can use `COPY --chown=someuser:somegroup /foo /bar` https://github.com/moby/moby/issues/6119#issuecomment-338920866

Comment (thasmo) [reply build/Dockerfile]: Oh, wow - that's actually perfect I guess. ❤️

**Comment by thasmo** [comment]
In terms of programming language, multi-stage builds enable us to use to use composition instead of inheritance. We can copy/use different/multiple images into a final image instead of just using a single image. Also, depending on how it is used, the final image is potentionally smaller.

**Comment by thasmo** [comment]
Also, I thought using `scratch` as base image helps to get the smallest possible image for `dst-academy/docker-dontstarvetogether-base` which only holds the game files. It's also faster to build. That means we can't use it as the base image for `dst-academy/docker-dontstarvetogether`, but that can be solved by just using a multi-stage build as well. It's just a prototype to find out if it works for us. So far I pretty much like it.

**Comment by thasmo** [comment]
Okay, it seems using an environment variable for `--chown` doesn't work. :/ Planned but not yet implemented. https://github.com/moby/moby/issues/35018

**Pull request closed without merging**